task/fips-operator-bundle-check/0.1/fips-operator-bundle-check.yaml

```diff
@@ -0,0 +1,136 @@
+            exit 0
+          else
+            echo "The label features.operators.openshift.io/fips-compliant is set to true. Running the FIPS static check..."
+          fi
+        else
+          echo "OpenShift Kubernetes Engine, OpenShift Platform Plus or OpenShift Container Platform are present in operators.openshift.io/valid-subscription. Running the FIPS static check..."
+        fi
+
+        unique_related_images=()
+        digests_processed=()
+        images_processed_template='{"image": {"pullspec": "'"$IMAGE_URL"'", "digests": [%s]}}'
+
+        echo "Inspecting raw image manifest $image_and_digest."
+        # Get the arch and image manifests by inspecting the image. This is mainly for identifying image indexes
+        image_manifests=$(get_image_manifests -i "${image_and_digest}")
+        echo "Image manifests are $image_manifests"
+
+        declare -A seen_related_images
+        # Extract relatedImages from the bundle image
+        while read -r _ arch_sha; do
+          digests_processed+=("\"$arch_sha\"")
+          bundle_render_out=$(opm render "$image_without_tag@$arch_sha")
+          manifest_related_images=$(echo "${bundle_render_out}" | jq -r '.relatedImages[]?.image')
+          if [ -n "$manifest_related_images" ]; then
+            for img in $manifest_related_images; do
+              if [ -z "${seen_related_images["$img"]:-}" ]; then
+                unique_related_images+=("$img")
+                seen_related_images["$img"]=1
+              fi
+            done
+          fi
+        done < <(echo "$image_manifests" | jq -r 'to_entries[] | "\(.key) \(.value)"')
+
+        echo "Unique related images: ${unique_related_images[*]}"
+        echo "${unique_related_images[*]}" > /tekton/home/unique_related_images.txt
+
+        # If the image is an Image Index, also add the Image Index digest to the list.
+        if [[ "${digests_processed[*]}" != *"$IMAGE_DIGEST"* ]]; then
+          digests_processed+=("\"$IMAGE_DIGEST\"")
+        fi
+        digests_processed_string=$(IFS=,; echo "${digests_processed[*]}")
+
+        echo "${images_processed_template/\[%s]/[$digests_processed_string]}" > /tekton/home/images_processed.txt
+
+    - name: fips-operator-check-step-action
+      computeResources:
+        limits:
+          memory: 512Mi
+          cpu: 200m
+        requests:
+          memory: 256Mi
+          cpu: 100m
+      ref:
+        name: fips-operator-check-step-action
+
+    - name: parse-images-processed-result
+      image: quay.io/redhat-appstudio/konflux-test:v1.4.9@sha256:eee855e60b437d9a55a30e63f2eb7f95d9fd6d3b111c32cac8730c9b7a071394
+      script: |
+        #!/usr/bin/env bash
+        set -euo pipefail
+
+        if [ -e "/tekton/home/images_processed.txt" ]; then
+          tee "$(results.IMAGES_PROCESSED.path)" < /tekton/home/images_processed.txt
+        else
+          echo "Task was skipped. Exiting"
+          exit 0
+        fi
```

Comment:
the idea behind this step is to only produce the IMAGES_PROCESSED result only if every other step was successful and not skipped. If one of the previous steps is skipped with `exit 0`, the file `/tekton/home/images_processed.txt` will not be present and hence, the task would not emit an IMAGES_PROCESSED result.

> I feel like we should move this up to before we exit early. Otherwise, if this task is required, we will hit EC validation failures for multi-arch builds as there will be unprocessed digests (even if the processing is determining that it needs to be skipped).

Yes, this task will end up as a required task.But, I don't think EC will complain if the IMAGES_PROCESSED result is not emitted in the first place. Based on [this](https://github.com/enterprise-contract/ec-policies/blob/main/antora/docs/modules/ROOT/pages/release_policy.adoc#291-image-digest-is-present-in-images_processed-result), EC only complains if the IMAGES_PROCESSED result is emitted and doesn't contain all the digests.

@lcarva Could you please correct me if I am mistaken with my EC assumption?